File: sdk/ios/common/GeotagData.swift
import HyperTrack

struct GeotagData {
    let data: [String: Any]
    let expectedLocation: HyperTrack.Location?
    let orderHandle: String?
    let orderStatus: HyperTrack.OrderStatus?

    init(
        data: [String: Any],
        expectedLocation: HyperTrack.Location?,
        orderHandle: String?,
        orderStatus: HyperTrack.OrderStatus?
    ) {
        self.data = data
        self.expectedLocation = expectedLocation
        self.orderHandle = orderHandle
        self.orderStatus = orderStatus
    }
}

File: sdk/android/src/main/java/com/reactnativehypertracksdk/common/GeotagData.kt
package com.reactnativehypertracksdk.common

import android.location.Location
import com.hypertrack.sdk.android.HyperTrack

/**
 * The data that represents geotag to create
 *
 * @param data Geotag payload
 */
internal data class GeotagData(
    val data: Map<String, Any?>,
    val expectedLocation: Location?,
    val orderHandle: String?,
    val orderStatus: HyperTrack.OrderStatus?,
)
